src/instructlab/chat/chat.py

```diff
@@ -478,6 +478,14 @@ def chat_cli(
         timeout=DEFAULT_CONNECTION_TIMEOUT,
         http_client=httpx.Client(cert=cert, verify=verify),
     )
+    # ensure the model specified exists on the server. with backends like vllm, this is crucial.
+    modelid_found = False
+    for api_model in client.models.list().data:
+        if api_model.id == model:
+            modelid_found = True
+
+    if not modelid_found:
+        raise ChatException(f"API issue found while executing chat, Model path {model} does not exist on the server.")
```

Comment:
```suggestion
        raise ChatException(f"Model path {model} does not exist on the server.")
```

Also, I'd print the list of available models to help the user further. Thanks!

Reply:
Might read better as: "Model {model} is not served by the server."?